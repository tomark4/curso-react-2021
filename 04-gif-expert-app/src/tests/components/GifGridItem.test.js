import GifGridItem from '../../components/GifGridItem';
import {shallow} from 'enzyme';

describe('Pruebas en GifGridItem', () => {
  const url = 'http://localhost/hola.jpg';
  const title = 'hola mundo';
  const wrapper = shallow( <GifGridItem url={url} title={title}/>);

  test('should show component correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should tener un parrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe( title );
  });

  test('should tener la url de imagen igual a las props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should tener la clase animate__bounce', () => {
    const div = wrapper.find('.card')
    expect(div.hasClass('animate__bounce')).toBe(true);
  });

});

