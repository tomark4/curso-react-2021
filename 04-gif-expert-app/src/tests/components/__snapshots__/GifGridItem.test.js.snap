// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem should show component correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="hola mundo"
    src="http://localhost/hola.jpg"
  />
  <p>
    hola mundo
  </p>
</div>
`;
